src/test/librbd/mirror/snapshot/test_mock_CreatePrimaryRequest.cc

```diff
@@ -166,11 +166,14 @@ class TestMockMirrorSnapshotCreatePrimaryRequest : public TestMockFixture {
 
   void expect_unlink_peer(MockTestImageCtx &mock_image_ctx,
                           MockUnlinkPeerRequest &mock_unlink_peer_request,
-                          uint64_t snap_id, const std::string &peer_uuid,
-                          int r) {
+                          uint64_t snap_id, std::list<std::string> &peer_uuids,
+                          int r, bool no_peer_test = false) {
     EXPECT_CALL(mock_unlink_peer_request, send())
-      .WillOnce(Invoke([&mock_image_ctx, &mock_unlink_peer_request, snap_id,
-                        peer_uuid, r]() {
+      .Times(peer_uuids.size())
+      .WillRepeatedly(Invoke([&mock_image_ctx, &mock_unlink_peer_request,
+                              snap_id, &peer_uuids, r, no_peer_test]() {
+                         auto peer_uuid = peer_uuids.front();
+                         peer_uuids.pop_front();
```

Comment:
I know @trociny suggested the list and it is indeed better than a static variable, but why not a simple `WillOnce` expectation added twice, once with "uuid1" and once with "uuid2"?  `InSequence` is used, so it should have the same effect.